script.js

```diff
@@ -0,0 +1,240 @@
+// getting the elements from the page
+const recipesContainer = document.querySelector('.recipesContainer');
+const recipesSearchResults = document.querySelector('.search-results');
+const recipeName = document.querySelector('.recipe_name');
+const recipeImg = document.querySelector('.recipeImg');
+const ingredientsList = document.querySelector('.ingredientsList');
+const cookingSteps = document.querySelector('.cookingSteps');
+const ingredientNumber = document.querySelector('.number-seach-input');
+const keywordInput = document.querySelector('.keyword-seach-input');
+const keywordSearchBtn = document.querySelector('.keyword-search-btn');
+const ingredientSearchBtn = document.querySelector('.ingr-nbr-find-btn');
+const timeCounterContainer = document.querySelector('.time-counter');
+const timeCount = document.querySelector('.time-count');
+const durationUnit = document.querySelector('.duration-unit')
+const timerContainer = document.querySelector('.timer-container');
+const minutesInput = document.querySelector('.time-amount-input');
+const setTimerBtn = document.querySelector('.set-timer-btn');
+const timerState = document.querySelector('.timer-state')
+
+//load previous added recipes from local storage, or initialize an empty list in case this is used for the first time
+const userRecipes = JSON.parse(localStorage.getItem("data")) || [];
+// the list of all recipes, will be an array of objects, to facilitate iteration and ordering
+//the list of units that are used in the recipes
+const unitTypes = {
+    weight: ["g", "kg"],
+    volume: ["ml", "dl", "l", "cup", "tbsp", "tsp"],
+    counts: ["piece", "dozen", "bottle", "pack", "clove"]
+
+    // shortcuts
+    // g = unitTypes.weight[0] kg = unitTypes.weight[1]
+    // ml = unitTypes.volume[0] ld =ml = unitTypes.volume[1] l = unitTypes.volume[2] cup = unitTypes.volume[3] tbsp = unitTypes.volume[4] tsp = unitTypes.volume[5] 
+    // pieces = unitTypes.counts[0] dozen = unitTypes.counts[1]   bottle = unitTypes.counts[2] pack = unitTypes.counts[3] clove = unitTypes.counts[4]
+};
+
+//getting userRecipes from local storage
+const userRecipesKey = "userRecipes"; // Key for local storage
+
+// Load user-created recipes from local storage
+const loadUserRecipes = () => {
+    const storedRecipes = localStorage.getItem(userRecipesKey);
+    return storedRecipes ? JSON.parse(storedRecipes) : [];
+};
+
+//consolidating all recipes in one list
+const getAllRecipes = () => {
+    const userRecipes = loadUserRecipes();
+    return [...builtInRecipes, ...userRecipes];
+  };
+
+//fetching the recipes and rendering them
+let builtInRecipes = []
+async function getRecipes() {
```

Comment:
try to add try catch to avoid having a failure when you receive an unexpected response or when you're unable to parse the output as json